kmk/scanners/digitalio.py

```diff
@@ -22,8 +22,9 @@ def __init__(
         rows,
         diode_orientation=DiodeOrientation.COL2ROW,
         pull=digitalio.Pull.UP,
-        rollover_cols_every_rows=None,
+        rollover_cols_every_rows=None,  # with value k, treat k*r x c matrix as r x c*k
         offset=0,
+        multiplexed=False,  # 2^k outputs are multiplexed on k output pins
```

Comment:
Not a big fan of packing even more conditional functionality into the already slow digitalio scanner.
I also don't quite understand what's happening here. Is that something that could be replaced by 
CircuitPythons native `keypad_demux.DemuxKeyMatrix` (currently beta)?

Reply:
I looked at `keypad_demux` docs, but it didn't seem to be available on my board yet.  It also didn't seem quite as flexible in terms of configuring which direction is multiplexed, and pullup v pulldown etc.

Honestly I was a bit confused why `digitialio.MatrixScanner` exists at all when we have `keypad.KeyMatrix` natively but I am pretty new to circuitpython/kmk.  I thought the modification was relatively small: in the regular case we scan column _n_ by activating the _n_-th column pin, but in the mutiplexed case we scan column _n_ by writing _n_ as a bit pattern _b0 b1 .. bk_ and set each multiplexed column bit.

If you prefer I could clone or subclass MatrixScanner to DemuxMatrixScanner instead?

Reply:
Getting `keypad_demux` build for your board shouldn't be too difficult. It is still in beta in any case.

The `digitalio` scanner predates the CP native matrix scanner, and there're hardware implementations that don't work with `KeyMatrix` -- at least for now -- that's the reason it's still around. If in doubt: always use the native scanner.
Thank you for the explanation, I understand what kind of mux we're talking about now.

Can you subclass without duplicating all of the code? I'm a bit short on time atm. and can't estimate that right now.